scripts/manage_demux_stats.py

```diff
@@ -89,76 +90,122 @@ def manipulate_process(demux_process, proc_stats):
     #Checks for document version
     if not "Document Version" in demux_process.udf:
         problem_handler("exit", "No Document Version set. Please set one.")
         
     try:
         demux_process.put()
-    except:
-        problem_handler("exit", "Failed to apply process thresholds to LIMS")
+    except Exception as e:
+        problem_handler("exit", "Failed to apply process thresholds to LIMS: {}".format(e.message))
     
 """Sets artifact = samples values """
 def set_sample_values(demux_process, parser_struct, proc_stats):
+    failed_entries = 0
+    undet_included = False
+    
     for pool in demux_process.all_inputs():
+        undet_reads = 0
+        undet_pairs = 0
+        assign_lane_reads = 0
+        undet_lane_reads = 0
+        
         try:
             outarts_per_lane = demux_process.outputs_per_input(pool.id, ResultFile = True)
-        except:
-            problem_handler("exit", "Unable to fetch artifacts of process")
+        except Exception as e:
+            problem_handler("exit", "Unable to fetch artifacts of process: {}".format(e.message))
         if proc_stats["Instrument"] == "miseq":
             lane_no = "1"
         else:
             try:
                 lane_no = pool.location[1][0]
-            except:
-                problem_handler("exit", "Unable to determine lane number. Incorrect location variable in process.")
+            except Exception as e:
+                problem_handler("exit", "Unable to determine lane number. Incorrect location variable in process: {}".format(e.message))
         logger.info("Lane number set to {}".format(lane_no))
         exp_smp_per_lne = round(demux_process.udf["Minimum Reads per Lane"]/float(len(outarts_per_lane)), 0)
         logger.info("Expected sample clusters for this lane: {}".format(str(exp_smp_per_lne)))
         
-        assign_lane_reads = 0
-        undet_lane_reads = 0
         for target_file in outarts_per_lane:
             try:
                 current_name = target_file.samples[0].name
-            except:
-                problem_handler("exit", "Unable to determine sample name. Incorrect sample variable in process.")
+            except Exception as e:
+                problem_handler("exit", "Unable to determine sample name. Incorrect sample variable in process: {}".format(e.message))
             for entry in parser_struct:
                 if lane_no == entry["Lane"]:
+                    lane_no = int(lane_no)
                     sample = entry["Sample"]
+                    
+                    if "Lanes to include undetermined" in demux_process.udf:
+                        try:
+                            undet_lanes= re.split('[ ,.]', demux_process.udf["Lanes to include undetermined"])
+                            undet_lanes = [int(i) for i in undet_lanes]
+                        except:
+                            problem_handler("exit", "Unable to typecast included undetermined lanes. Possibly non-number in list")
```

Comment:
Why does this block need to be repeated for each entry ? Can't you set the lanes to include undetermined once for all entries ?